src/Void.Core/BufferMessages.fs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Void.Core
+
+type BufferMessage = inherit Message
+
+type BufferEnvelopeMessage<'TBufferMessage when 'TBufferMessage :> BufferMessage> =
+    {
+        BufferId : int
+        Message : 'TBufferMessage
+    }
+    interface EnvelopeMessage<'TBufferMessage>
+
+type FileBufferProxy = {
+    MaybeFilepath : string option
+    Contents : string seq
+}
+
+[<RequireQualifiedAccess>]
+type BufferCommand =
+    | MoveCursor of Motion
+    interface CommandMessage
+    interface BufferMessage
+
+[<RequireQualifiedAccess>]
+type BufferEvent =
+    | Added of FileBufferProxy

```

Comment:
No, it means "a (file) buffer was added". Maybe I should have spelled out the name more. I was reading it like `Buffer(Event).Added`.